aten/src/ATen/native/Linear.cpp

```diff
@@ -641,9 +642,26 @@ Tensor tensordot(const Tensor& input1, const Tensor& input2, IntArrayRef dims1,
 }
 
 Tensor &tensordot_out(const Tensor& input1, const Tensor& input2, IntArrayRef dims1, IntArrayRef dims2, Tensor& result) {
-  result.copy_(at::native::tensordot(input1, input2, dims1, dims2));
+  Tensor result_tmp = at::native::tensordot(input1, input2, dims1, dims2);
+  auto result_dtype = result_tmp.scalar_type();
+  auto output_tensor_dtype = result.scalar_type();
+  auto output_device = result.device();
+  auto input_device = input1.device();
+  // check if the input & output tensors are on the same device.
+  TORCH_CHECK(
+    output_device == input_device,
+    "tensordot_out: Expected the output and input tensors to be on the "
```

Comment:
This is great but let's just use the user-facing name for error messages, in this case "tensordot" instead of "tensordot_out"

Reply:
Thanks, I'll do so when the currently running CI checks would complete.